ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js

```diff
@@ -0,0 +1,155 @@
+  } = useUserPreferencedCurrency(PRIMARY);
+  const {
+    currency,
+    numberOfDecimals: numberOfDecimalsFiat,
+  } = useUserPreferencedCurrency(SECONDARY);
+
+  const advancedGasFeeValues = useSelector(getAdvancedGasFeeValues);
+
+  const [editingInGwei, setEditingInGwei] = useState(false);
+
+  const [maxBaseFeeGWEI, setMaxBaseFeeGWEI] = useState(() => {
+    if (
+      estimateUsed !== PRIORITY_LEVELS.CUSTOM &&
+      advancedGasFeeValues?.maxBaseFee
+    ) {
+      return multiplyCurrencyValues(
+        estimatedBaseFee,
+        advancedGasFeeValues.maxBaseFee,
+        numberOfDecimalsPrimary,
+      );
+    }
+    return maxFeePerGas;
+  });
+
+  const [maxBaseFeeMultiplier, setMaxBaseFeeMultiplier] = useState(() => {
+    if (
+      estimateUsed !== PRIORITY_LEVELS.CUSTOM &&
+      advancedGasFeeValues?.maxBaseFee
+    ) {
+      return advancedGasFeeValues.maxBaseFee;
+    }
+    return divideCurrencyValues(maxFeePerGas, estimatedBaseFee);
+  });
+
+  const [, { value: baseFeeInFiat }] = useCurrencyDisplay(
+    decGWEIToHexWEI(maxBaseFeeGWEI),
+    { currency, numberOfDecimalsFiat },
+  );
+
+  const updateBaseFee = useCallback(
+    (value) => {
+      let baseFeeInGWEI;
+      let baseFeeMultiplierValue;
+      if (editingInGwei) {
+        baseFeeInGWEI = value;
+        baseFeeMultiplierValue = divideCurrencyValues(value, estimatedBaseFee);
+      } else {
+        baseFeeInGWEI = multiplyCurrencyValues(
+          estimatedBaseFee,
+          value,
+          numberOfDecimalsPrimary,
+        );
+        baseFeeMultiplierValue = value;
+      }
+      setMaxBaseFeeGWEI(baseFeeInGWEI);
+      setMaxBaseFeeMultiplier(baseFeeMultiplierValue);
+      setDirty(true);
+    },
+    [
+      editingInGwei,
+      estimatedBaseFee,
+      numberOfDecimalsPrimary,
+      setMaxBaseFeeGWEI,
+      setMaxBaseFeeMultiplier,
+      setDirty,
+    ],
+  );
+
+  useEffect(() => {
+    setMaxFeePerGas(maxBaseFeeGWEI);
+  }, [maxBaseFeeGWEI, setMaxFeePerGas]);
+
+  return (
+    <Box className="base-fee-input">
+      <FormField
+        onChange={updateBaseFee}
+        titleText={t('maxBaseFee')}
+        titleUnit={editingInGwei ? 'GWEI' : `(${t('multiplier')})`}
+        tooltipText={t('advancedBaseGasFeeToolTip')}
+        titleDetail={
+          <Button
+            className="base-fee-input__edit-link"
+            type="link"
+            onClick={() => setEditingInGwei(!editingInGwei)}
+          >
+            <I18nValue
+              messageKey={editingInGwei ? 'editInMultiplier' : 'editInGwei'}
+            />
+          </Button>
+        }
+        value={editingInGwei ? maxBaseFeeGWEI : maxBaseFeeMultiplier}
+        detailText={
+          editingInGwei
+            ? `${maxBaseFeeMultiplier}x ${`≈ ${baseFeeInFiat}`}`
+            : `${maxBaseFeeGWEI} GWEI ${`≈ ${baseFeeInFiat}`}`
+        }
+        numeric
+      />
+      <AdvancedGasFeeInputSubtext
+        latest={estimatedBaseFee}
+        historical="23-359 GWEI"
```

Comment:
Presuming this will be calculated in a follow up PR.

Reply:
yes